src/server/models/listItem.ts

```diff
@@ -195,6 +204,143 @@ export async function checkListItemExists({
   return total > 0;
 }
 
+export async function getPaginationValues(  props: {
+  count: number;
+  page: number;
+  params: ListsRequestParams;
+}): Promise<PaginationResults> {
+  const { count, page, params } = props;
+  const limit = 20;
+  let allPages = 0;
+  if (count > 0 && limit > 0) {
+    allPages = Math.ceil(count / limit);
+  }
+
+  const nextPrevious = await getNextPrevious({page, allPages, params});
+  const { queryString, currentPage } = nextPrevious;
+
+  const pageItems: PaginationItem[] = await getPageItems({allPages, currentPage, queryString});
+
+  const from = await getFromCount({count, limit, currentPage});
+  const to = await getToCount({currentPage, allPages, count, limit});
+
+  return {
+    pagination: {
+      results: {
+        from,
+        to,
+        count,
+        currentPage
+      },
+      previous: {
+        text: nextPrevious.previous.page.toString(),
+        href: nextPrevious.previous.queryString
+      },
+      next: {
+        text: nextPrevious.next.page.toString(),
+        href: nextPrevious.next.queryString
+      },
+      items: pageItems
+    }
+  }
+}
+
+ async function getNextPrevious(props: {page: number; allPages: number; params: ListsRequestParams }): Promise<{
+  queryString: string;
+  currentPage: number;
+  previous: {
+    page: number;
+    queryString: string;
+  };
+  next: {
+     page: number;
+     queryString: string;
+   };
+ }> {
+  const {page, allPages, params} = props;
+  let currentPage = page === undefined ? 1 : Number(page);
+  let queryStringPrevious = "";
+  let queryStringNext = "";
+  let previousPage = -1;
+  let nextPage = -1;
+
+  let queryString = await queryStringFromParams(params, true);
+  queryString = queryString.replace("&page=" + currentPage.toString(), "");
```

Comment:
if there's nothing stopping us from using querystring, we might as well use it. the node class can parse it into an object for us and back again. I always find search/replace a bit jank

```suggestion
  let query = querystring.parse(params)
  query.page = currentPage 
  const stringified = querystring.stringify(query)
```